applications/shape_detector/src/plane_detector.py

```diff
@@ -0,0 +1,939 @@
+
+        elif img_type == 7: # stair
+
+            x,y             = np.meshgrid(np.arange(w),np.arange(h))   
+            self.img        = (np.sign(x - w/2) + np.sign(y - h/2))*5 + 200 # less slope     
+
+
+        elif img_type == 8: # corner
+
+            x,y             = np.meshgrid(np.arange(w),np.arange(h))   
+            self.img        = np.ones((h,w))*250
+            img_bool        = np.logical_and((x - w/2) < 0, (y - h/2) < 0)
+            self.img[img_bool] = 230 # quarter                            
+
+        elif img_type == 10: # flat
+
+            self.img        = np.ones((h,w))*500             
+
+        elif img_type == 11:
+            "chess board"
+            fname = r"C:\Users\udubin\Documents\Code\opencv-4x\samples\data\left04.jpg"
+            self.img        = cv.imread(fname)
+
+        elif img_type == 12:
+            self.img = cv.imread('image_scl_001.png', cv.IMREAD_GRAYSCALE)
+            self.img = cv.resize(self.img , dsize = self.frame_size) 
+            
+        elif img_type == 13:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\image_ddd_000.png", cv.IMREAD_GRAYSCALE)
+            self.img = cv.resize(self.img , dsize = self.frame_size) 
+
+        elif img_type == 14:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\image_ddd_001.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)     
+
+        elif img_type == 15:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\image_ddd_002.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)     
+
+        elif img_type == 16:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\image_ddd_003.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)   
+
+        elif img_type == 17:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\floor_view_default_Depth_Depth.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)          
+
+        elif img_type == 18:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\floor_view_default_corner2_Depth_Depth.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)                   
+
+        elif img_type == 21:
+            self.img = cv.imread(r"C:\Data\Depth\Plane\image_scl_000.png", cv.IMREAD_GRAYSCALE)  
+            self.img = cv.resize(self.img , dsize = self.frame_size)                                     
+            
+        #self.img        = np.uint8(self.img) 
+
+        self.img = self.add_noise(self.img, 0)
+        self.frame_size = self.img.shape[:2]      
+        return self.img
+      
+    def init_roi(self, test_type = 1):
+        "load the test case"
+        roi = [0,0,self.frame_size[0],self.frame_size[1]]
+        if test_type == 1:
+            roi = [310,230,330,250] # xlu, ylu, xrb, yrb
```

Comment:
can a comment be added to give a hint to the reader about the meaning of various rois